### PR TITLE
add run job capabilities

### PR DESCRIPTION
**Pull request by billiford**
- adds capabilities for stage Run Job (Manifest)
- allows for custom run jobs as well

### CODE REVIEW

File: pkg/kubernetes/job_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+				Expect(s).To(Equal("Running"))
+			})
+		})
+
+		When("the job succeeded", func() {
+			BeforeEach(func() {
+				completions := int32(1)
+				o := job.Object()
+				o.Status.CompletionTime = &metav1.Time{}
+				o.Status.Succeeded = int32(1)
```
Comment (yasmin-abdullah) [inline pkg/kubernetes/job_test.go:76]: This looks different than the pervious test 
```
o.Status.Conditions = []v1.JobCondition{
					{
						Type: "Failed",
					},
				}
```
isn't there something similar like Type: "Succeeded"?

**Review by yasmin-abdullah — CHANGES_REQUESTED** [review]

File: pkg/http/core/applications.go [hunk 2]
```diff
@@ -975,3 +975,93 @@ func GetServerGroup(c *gin.Context) {
...
+	ExitCode string `json:"exitCode"`
+	Message  string `json:"message"`
+	Reason   string `json:"reason"`
+	Signal   string `json:"signal"`
+}
+
+func GetJob(c *gin.Context) {
+	sc := sql.Instance(c)
+	kc := kubernetes.ControllerInstance(c)
+	ac := arcade.Instance(c)
```
[the remark this answers is not in the document]
Comment (billiford) [reply pkg/http/core/applications.go:1000]: ac = arcade client, sc = sql client, kc = kubernetes controller.

File: pkg/http/core/applications.go [hunk 3]
```diff
@@ -975,3 +975,93 @@ func GetServerGroup(c *gin.Context) {
...
+	job := Job{
+		Account: account,
+		CompletionDetails: JobCompletionDetails{
+			ExitCode: "",
+			Message:  "",
+			Reason:   "",
+			Signal:   "",
+		},
+		CreatedTime: result.GetCreationTimestamp().Unix() * 1000,
+		JobState:    j.State(),
```
[the remark this answers is not in the document]
Comment (billiford) [reply pkg/http/core/applications.go:1059]: state != status. The state of the job is defined [here](https://github.com/billiford/go-clouddriver/blob/runjob/pkg/kubernetes/job.go#L39) - status is a TODO.

The state of the job is calculated for a `runJob`. The status of a job is calculated for a `deployManifest`.

File: pkg/kubernetes/job_test.go (hunk 1 above)
Comment (billiford) [reply pkg/kubernetes/job_test.go:76]: Checking the conditions of a job is different than checking the number of succeeded operations.

I followed the [source code](https://github.com/spinnaker/clouddriver/blob/master/clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/model/KubernetesJobStatus.java#L79) on this one, so we should be good.

**Review by yasmin-abdullah — APPROVED** [review]

**Pull request merged**